src/url.rs

```diff
@@ -48,48 +48,38 @@ pub struct HttpUrl {
 }
 
 impl HttpUrl {
-    pub fn port(&self) -> Result<u16> {
-        match self.url.port_or_known_default() {
-            Some(v) => Ok(v),
-            None => Err(Error::UrlError(format!(
-                "unsupported URL scheme {}: no port and no known default",
-                self.url.scheme()
-            ))),
-        }
+    pub fn port(&self) -> u16 {
+        // this should always go to the happy path as we only have scheme of http and https
```

Comment:
I would phrase this as "this will never fail because we verified the scheme is HTTP or HTTPS which should always have a port"